pilot/pkg/proxy/envoy/v2/discovery_test.go

```diff
@@ -295,3 +295,76 @@ func TestDebounce(t *testing.T) {
 		})
 	}
 }
+
+func TestDebounceSyncPush(t *testing.T) {
+	// This test tests the timeout and debouncing of config updates
+	// If it is flaking, DebounceAfter may need to be increased, or the code refactored to mock time.
+	// For now, this seems to work well
+	DebounceAfter = time.Millisecond * 50
+	DebounceMax = DebounceAfter * 2
+
+	tests := []struct {
+		name string
+		test func(updateCh chan *model.PushRequest, expect func(partial, full int32))
+	}{
+		{
+			name: "Should push synchronously after debounce",
+			test: func(updateCh chan *model.PushRequest, expect func(partial, full int32)) {
+				updateCh <- &model.PushRequest{Full: true}
+				time.Sleep(DebounceAfter)
```

Comment:
This looks like a close call here.
I would suggest giving it some additional time, so the test won't be flaky (especially when running on a busy ci-machine, which runs all tests in parallel). 
```suggestion
				time.Sleep(DebounceAfter + 10*time.Millisecond)
```